launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +121,30 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   *
+   * @since 2.2.0
+   * @return This launcher.
+   */
+  public SparkLauncher stopIfLauncherShutdown() {
```

Comment:
BTW, one thing you can do here is to change this to something like `autoShutdown`, which just means that the app will try to shut itself down if the parent process disconnects from the app. (So it's not just restricted to "shutdown", since a "disconnect" would also trigger it.)

Then you can document in "disconnect()" that this option affects its behavior. Since this behavior is opt-in, it should be ok.